app/javascript/packs/components/forms/LocationInput.jsx

```diff
@@ -0,0 +1,165 @@
+import React from "react";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import LocationOnIcon from "@material-ui/icons/LocationOn";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/core/styles";
+import parse from "autosuggest-highlight/parse";
+import throttle from "lodash/throttle";
+import StyledTextField from "./StyledTextField";
+
+function loadScript(src, position, id) {
+  if (!position) {
+    return;
+  }
+
+  const script = document.createElement("script");
+  script.setAttribute("async", "");
+  script.setAttribute("id", id);
+  script.src = src;
+  position.appendChild(script);
+}
+
+const autocompleteService = { current: null };
+
+const geocoderService = { current: null };
+
+const useStyles = makeStyles((theme) => ({
+  icon: {
+    color: theme.palette.text.secondary,
+    marginRight: theme.spacing(2),
+  },
+}));
+
+export default function LocationInput({
+  value,
+  setAddress,
+  inputProps,
+  onChange,
+}) {
+  const classes = useStyles();
+  const [inputValue, setInputValue] = React.useState("");
+  const [options, setOptions] = React.useState([]);
+  const loaded = React.useRef(false);
+
+  if (typeof window !== "undefined" && !loaded.current) {
+    if (!document.querySelector("#google-maps")) {
+      loadScript(
+        `https://maps.googleapis.com/maps/api/js?key=${process.env.GOOGLE_MAPS_API_KEY}&libraries=places`,
```

Comment:
Given that this is going to be exposed to the user anyway, I think you can either hardcode a key, or we can pass it to the frontend with a request to something like `/env/variables.js`, served by a new Rails controller, that sets some variables on `window`.

Reply:
Google Maps API keys can be domain-locked so that sharing them isn't harmful.

Reply:
Good call! I’ll add dotenv and put it in there.

Reply:
Maybe just commit it in the JS then? Since it's not sensitive? 🤷‍♂️